sdk/translation/azure-ai-translation-document/tests/test_all_document_statuses.py

```diff
@@ -4,126 +4,206 @@
+        docs_count = 5
+        target_language = "es"
+
+        # submit and validate job
+        job_id = self._create_translation_job_with_dummy_docs(client, docs_count, language_code=target_language, wait=True)
+
+        # check doc statuses
+        doc_statuses = list(client.list_all_document_statuses(job_id, order_by=["createdDateTimeUtc asc"])) # convert from generic iterator to list
+        self.assertEqual(len(doc_statuses), docs_count)
+
+        curr = datetime.min
+        for document in doc_statuses:
+            assert(document.created_on.replace(tzinfo=None) >= curr.replace(tzinfo=None))
+            curr = document.created_on
+
+
+    @DocumentTranslationPreparer()
+    @DocumentTranslationClientPreparer()
+    def test_list_document_statuses_order_by_creation_time_desc(self, client):
+        docs_count = 5
+        target_language = "es"
+
+        # submit and validate job
+        job_id = self._create_translation_job_with_dummy_docs(client, docs_count, language_code=target_language, wait=True)
+
+        # check doc statuses
+        doc_statuses = list(client.list_all_document_statuses(job_id, order_by=["createdDateTimeUtc desc"])) # convert from generic iterator to list
+        self.assertEqual(len(doc_statuses), docs_count)
+
+        curr = datetime.max
+        for document in doc_statuses:
+            assert(document.created_on.replace(tzinfo=None) <= curr.replace(tzinfo=None))
+            curr = document.created_on
+
+
+    @pytest.mark.skip(reason="not working! - list returned is empty")
+    @DocumentTranslationPreparer()
+    @DocumentTranslationClientPreparer()
+    def test_list_document_statuses_mixed_filters(self, client):
+        docs_count = 25
+        target_language = "es"
+        skip = 3
+        results_per_page = 2
+        statuses = ["Succeeded"]
+
+        # submit and validate job
+        job_id = self._create_translation_job_with_dummy_docs(client, docs_count, language_code=target_language, wait=True)
+
+        # get ids
+        doc_statuses = list(client.list_all_document_statuses(job_id)) # convert from generic iterator to list
+        self.assertEqual(len(doc_statuses), docs_count)
+        ids = [doc.id for doc in doc_statuses]
+        ids = ids[:docs_count//2]
+
+        # list jobs
+        filtered_docs = client.list_all_document_statuses(
+            job_id,
+            # filters
+            document_ids=ids,
+            statuses=statuses,
+            # ordering
+            order_by=["createdDateTimeUtc asc"],
+            # paging
+            skip=skip,
+            results_per_page=results_per_page
+        ).by_page()
+        self.assertIsNotNone(filtered_docs)
+
+        # check statuses
+        counter = 0
+        curr_time = datetime.max
+        for page in filtered_docs:
+            page_docs = list(page)
+            self.assertLessEqual(len(page_docs), results_per_page) # assert paging
+            for doc in page:
```

Comment:
Since we convert to list on L198 we already exhausted calling the nextLinks. There won't be anything here to iterate on (my guess for why the list is returning empty). You can just iterate over the list you created.

```suggestion
            for doc in page_docs:
```

Reply:
yeah sorry missed that!
but still not working